### PR TITLE
Option for PowerDnsProvider to request notification after changes

### PR DESCRIPTION
**Pull request by hanazuki**
Hi. We are running pdns as our primary nameserver and our secondary servers replicate the zones from the primary with AXFR. While pdns periodically (once per several minutes, I suppose) checks if a zone is modified and sends DNS NOTIFY to the secondaries, we need more prompt propagation of changes, which can be achieved by `PUT /zones/{zone_id}/notify` API.

This patch introduces `notify` option to PowerDnsProvider, which instructs the provider to ask the pdns server for sending a notification to its secondary servers after changing zone content. The option is off by default since the notification request will fail unless the pdns server operates as a primary for the zone.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,6 +48,8 @@ providers:
       # scheme: https
       # Check SSL certificate (optional, default True)
       # ssl_verify: true
+      # Send DNS NOTIFY to secondary servers after change (optional, default false)
+      notify: true
```
Comment (ross) [inline README.md:52]: ~convention here is to put the default value in the example.

```suggestion
      notify: false
```

**Review by ross — COMMENTED** [review]
Looks good. Didn't know this was a thing with PowerDNS. I suspect it'll need test coverage to pass CI. Thanks for the find & PR.

**Comment by hanazuki** [comment]
Thank you for reviewing my proposal. Will work on a test case.

**Comment by ross** [comment]
> Will work on a test case.

👍 lmk if you have any questions or run into any problems.

**Comment by hanazuki** [comment]
Updated the patch to include a test case for the feature introduced. I believe the case not calling /notify API when notify=False is covered by the existing test cases. Please take a look.

**Review by ross — APPROVED** [review]

**Comment by ross** [comment]
Thanks!

**Pull request merged**